android-test/src/androidTest/java/okhttp/android/test/OkHttpTest.kt

```diff
@@ -196,8 +196,14 @@ class OkHttpTest {
         assertEquals(200, response.code)
         // see https://github.com/google/conscrypt/blob/b9463b2f74df42d85c73715a5f19e005dfb7b802/android/src/main/java/org/conscrypt/Platform.java#L613
         when {
+            // TODO: how do we determine when we expect useEngineSocket to be true, resulting in org.conscrypt.Java8EngineSocket?
```

Comment:
this is maybe not worth doing, if the below or similar if branch is ok. Alternatively, we can switch to assertJ and use a startsWith assertion

Reply:
Are you running these as normal JDK tests in Intellij?  They should hopefully be running on device or in an emulator.

Reply:
sorry I guess I had a pretty horrible comment there. I will put on the test itself that this is going to execute in the emulator and strip notions of JDK.

meanwhile my version of pixel5 fails on this unless I permit this to resolve to Java8EngineSocket. This is exactly the cognitive thing I was hoping you could help with.. is there a way to tell as we do in the branch on `Build.VERSION.SDK_INT` whether this will be Java8EngineSocket or Java8FileDescriptorSocket? I looked at the code and it seems to be effected by a flag `useEngineSocket`. FWIW you can see my emulator is using sdk 29..

I'll update the comments after you teach me more :D thanks for the help

Reply:
I think this shows a bug in the test setup previously, sorry.  I think we just want org.conscrypt.Java8EngineSocket here.

https://groups.google.com/g/conscrypt/c/Ajrexsz_BQc

```
assertEquals("org.conscrypt.Java8EngineSocket", socketClass)
```